packages/core-cairo/src/api.ts

```diff
@@ -0,0 +1,31 @@
+import type { CommonOptions } from './common-options';
+import { printERC20, defaults as erc20defaults, ERC20Options } from './erc20';
+import { printERC721, defaults as erc721defaults, ERC721Options } from './erc721';
+
+export interface WizardContractAPI {
+  /**
+   * Returns a string representation of a contract generated using the provided options. If opts is not provided, uses `defaults`.
+   */
+  print: (opts?: any) => string,
+  
+  /**
+   * The default options that are used for `print`.
+   */
+  defaults: CommonOptions;
+}
+
+export interface ERC20 extends WizardContractAPI {
+  defaults: Required<ERC20Options>;
+}
+export interface ERC721 extends WizardContractAPI {
+  defaults: Required<ERC721Options>;
+}
```

Comment:
I believe this encodes what we want better:

```suggestion
export interface WizardContractAPI<Options extends CommonOptions> {
  /**
   * Returns a string representation of a contract generated using the provided options. If opts is not provided, uses `defaults`.
   */
  print: (opts?: Options) => string,
  
  /**
   * The default options that are used for `print`.
   */
  defaults: Required<Options>;
}

export type ERC20 = WizardContractAPI<ERC20Options>;
export type ERC721 = WizardContractAPI<ERC721Options>;
```